pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java

```diff
@@ -198,17 +198,20 @@ public interface ConsumerBuilder<T> extends Cloneable {
     ConsumerBuilder<T> ackTimeout(long ackTimeout, TimeUnit timeUnit);
 
     /**
-     * Acknowledgement returns receipt, but the message is not re-sent after getting receipt.
+     * Enables or disables the acknowledgment receipt feature.
```

Comment:
I think that it is worth to note that even if the server acknowledged the operation, this method doesn't guarantee that the message won't be redelivered.
Because on the broker side acknowledgement are persisted periodically or by other strategies (no need to explain here)

Reply:
Thanks, I'll mention that.

Reply:
@eolivelli added "Even when the broker returns a receipt, it doesn't guarantee that the message won't be redelivered later due to certain implementation details."